src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -0,0 +1,266 @@
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Cache.Memory;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="TagHelper"/> implementation targeting &lt;cache&gt; elements.
+    /// </summary>
+    public class CacheTagHelper : TagHelper
+    {
+        private const string VaryByAttributeName = "asp-vary-by";
+        private const string VaryByHeaderAttributeName = "asp-vary-by-header";
+        private const string VaryByQueryAttributeName = "asp-vary-by-query";
+        private const string VaryByRouteAttributeName = "asp-vary-by-route";
+        private const string VaryByCookieAttributeName = "asp-vary-by-route";
+        private const string VaryByUserAttributeName = "asp-vary-by-user";
+        private const string ExpiresOnAttributeName = "asp-expires-on";
+        private const string ExpiresAfterAttributeName = "asp-expires-after";
+        private const string ExpiresSlidingAttributeName = "asp-expires-sliding";
+        private const string CachePriorityAttributeName = "asp-priority";
+        private const string CacheKeyTokenSeparator = "||";
+        private static readonly char[] AttributeSeparator = new[] { ',' };
+
+        [Activate]
+        public IMemoryCache MemoryCache { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ViewContext"/> for the current executing View.
+        /// </summary>
+        [Activate]
+        public ViewContext ViewContext { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="string" /> to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByAttributeName)]
+        public string VaryBy { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of a HTTP request header to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByHeaderAttributeName)]
+        public string VaryByHeader { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of query parameters to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByQueryAttributeName)]
+        public string VaryByQuery { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of route data parameters to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByRouteAttributeName)]
+        public string VaryByRoute { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of cookie names to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByRouteAttributeName)]
+        public string VaryByCookie { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines if the content is to be varied by the Identity for the logged in 
+        /// <see cref="HttpContext.User"/>.
```

Comment:
nit: other summaries use "to vary the cached result by".  should probably be consistent here, at least using "cached result" and not "content".